lib/libutee/arch/arm/gprof/gprof.h

```diff
@@ -0,0 +1,256 @@
+ *
+ * For example, on the VAX, the shortest two call sequence is:
+ *
+ *	calls	$0,(r0)
+ *	calls	$0,(r0)
+ *
+ * which is separated by only three bytes, thus HASHFRACTION is
+ * calculated as:
+ *
+ *	HASHFRACTION = 3 / (2 * 2 - 1) = 1
+ *
+ * Note that the division above rounds down, thus if MIN_SUBR_FRACTION
+ * is less than three, this algorithm will not work!
+ *
+ * In practice, however, call instructions are rarely at a minimal
+ * distance.  Hence, we will define HASHFRACTION to be 2 across all
+ * architectures.  This saves a reasonable amount of space for
+ * profiling data structures without (in practice) sacrificing
+ * any granularity.
+ */
+#define	HASHFRACTION	2
+
+/*
+ * Percent of text space to allocate for tostructs.
+ * This is a heuristic; we will fail with a warning when profiling programs
+ * with a very large number of very small functions, but that's
+ * normally OK.
+ * 2 is probably still a good value for normal programs.
+ * Profiling a test case with 64000 small functions will work if
+ * you raise this value to 3 and link statically (which bloats the
+ * text size, thus raising the number of arcs expected by the heuristic).
+ */
+#define ARCDENSITY	3
+
+/*
+ * Always allocate at least this many tostructs.  This
+ * hides the inadequacy of the ARCDENSITY heuristic, at least
+ * for small programs.
+ */
+#define MINARCS		50
+
+/*
+ * The type used to represent indices into gmonparam.tos[].
+ */
+#define	ARCINDEX	unsigned long
+
+/*
+ * Maximum number of arcs we want to allow.
+ * Used to be max representable value of ARCINDEX minus 2, but now
+ * that ARCINDEX is a long, that's too large; we don't really want
+ * to allow a 48 gigabyte table.
+ * The old value of 1<<16 wasn't high enough in practice for large C++
+ * programs; will 1<<20 be adequate for long?  FIXME
+ */
+#define MAXARCS		(1 << 20)
+
+struct tostruct {
+	unsigned long selfpc;
+	long count;
+	ARCINDEX link;
+};
+
+/*
+ * A raw arc, with pointers to the calling site and the called site and a count.
+ */
+struct rawarc {
+	unsigned long	raw_frompc;
+	unsigned long	raw_selfpc;
+	long		raw_count;
+};
+
+/*
+ * general rounding functions.
+ */
+#define ROUNDDOWN(x,y)	(((x)/(y))*(y))
+#define ROUNDUP(x,y)	((((x)+(y)-1)/(y))*(y))
+
+/*
+ * The profiling data structures are housed in this structure.
+ */
+struct gmonparam {
+	long int	state;
+	unsigned short	*kcount;
+	unsigned long	kcountsize;
+	ARCINDEX	*froms;
+	unsigned long	fromssize;
+	struct tostruct	*tos;
+	unsigned long	tossize;
+	unsigned long	tolimit;
+	unsigned long	lowpc;
+	unsigned long	highpc;
+	unsigned long	textsize;
+	unsigned long	hashfraction;
+	long		log_hashfraction;
+};
+
+/*
+ * Possible states of profiling.
+ */
+#define	GMON_PROF_ON	0
+#define	GMON_PROF_BUSY	1
+#define	GMON_PROF_ERROR	2
+#define	GMON_PROF_OFF	3
+
+/*
+ * gmon.out file format
+ *
+ * This part adapted from glibc:
+ *   gmon/sys/gmon_out.h
+ *
+ * Copyright (C) 1996-2016 Free Software Foundation, Inc.
```

Comment:
https://www.youtube.com/watch?v=JlIrSMzF8T4 , my understanding is that any company using OP-TEE are not enforced to give away the source for the major part of OP-TEE, but they must give away the LGPL code from OP-TEE. That might not sound too bad, but I can imagine there are companies using OP-TEE that don't want to deal with this at all. I.e, they don't want to spend time on picking subsets of OP-TEE in case someone asks for it.

Having that said, we could poll the known users of OP-TEE, but that is not a very good way of dealing with questions like this.

As usual, I'm unsure whether the license covers compiled code, or both that and the source code available in the branch (other branches cannot impact I hope). I you have experience in this area, please let me know.

Fallback? Put it on a separate branch as of now?